### ISSUE TITLE
GSSAPI Authentication (Enterprise) Doc Update - hba configuration example

### ISSUE THREAD

**Issue by fabiog1901** [cockroachdb/docs#10888]
Re: [GSSAPI Authentication (Enterprise)](https://cockroachlabs.com/docs/v21.1/gssapi_authentication.html)

## Issue Description

we don't have documentation on how to use a client key+crt to login into a kerberized database. In this case, a client app `app1` should be authenticated using key+crt instead of a krb ticket. 

So having created the crt+key, and passing them to the DBURL, CockroachDB refuses them as we miss a piece of configuration which is not documented.

Example: The keys are created and passed to the DBURL, but auth fails as it's checking for a krb ticket

```
ubuntu@cockroachdb-1:~$ sudo ls -l /var/lib/cockroach/certs/
total 32
-rwxr-xr-x 1 cockroach cockroach 1123 Jun 11 18:08 ca.crt
-r-------- 1 root      root      4034 Jun 11 18:23 client.app1.crt
-r-------- 1 root      root      1676 Jun 11 18:23 client.app1.key
-r-------- 1 cockroach cockroach 4033 Jun 11 18:08 client.root.crt
-r-------- 1 cockroach cockroach 1679 Jun 11 18:08 client.root.key
-rwxr-xr-x 1 cockroach cockroach 4583 Jun 11 18:08 node.crt
-r-------- 1 cockroach cockroach 1675 Jun 11 18:08 node.key
ubuntu@fabio-demo-0-cockroachdb-1-45596821-0:~$ 
ubuntu@fabio-demo-0-cockroachdb-1-45596821-0:~$ sudo cockroach sql --url "postgresql://app1@<hostname>:26257/defaultdb?sslmode=require&sslrootcert=/var/lib/cockroach/certs/ca.crt&sslcert=/var/lib/cockroach/certs/client.app1.crt&sslkey=/var/lib/cockroach/certs/client.app1.key" 
#
# Welcome to the CockroachDB SQL shell.
# All statements must be terminated by a semicolon.
# To exit, type: \q.
#
ERROR: pq: kerberos error: open /tmp/krb5cc_0: no such file or directory
Failed running "sql"
```

## Suggested Resolution 

This page offers a ton of help and we should source a lot from it: https://dr-knz.net/authentication-in-postgresql-and-cockroachdb.html

In short, we should add steps required to setup the HBA configuration so that `app1` bypasses kerberos:

```
> SET cluster setting server.host_based_authentication.configuration = 'host all app1 all cert-password
host all all all gss include_realm=0';
> show cluster setting server.host_based_authentication.configuration;                                                                                                                                                                      
  server.host_based_authentication.configuration
--------------------------------------------------
  host all app1 all cert-password
  host all all all gss include_realm=0
(1 row)
```
